### PR TITLE
WIP: Support for Arc UI to use mesh-models client

### PR DESCRIPTION
**Pull request by kraman**
Connect to strongloop-internal/scrum-nodeops#331

### CODE REVIEW

**Comment by ritch** [comment]
Code looks good... not sure why the build is failing.

File: proxy/build-client.js [hunk 1]
```diff
@@ -25,6 +27,20 @@ module.exports = function buildBrowserBundle(env, out, callback) {
     return callback(err);
   }
 
+  try {
+    boot.compileToBrowserify({
+      appId: 'meshClient',
```
Comment (ritch) [inline proxy/build-client.js]: Did we end up using `appId`...? Is this merged?

Comment (kraman) [inline proxy/build-client.js]: yes

**Comment by ritch** [comment]
Looks good :+1:

**Comment by kraman** [comment]
@slnode test please

**Comment by kraman** [comment]
@slnode test please

**Comment by kraman** [comment]
@slnode test please

**Comment by kraman** [comment]
@ritch test failures seem to be unrelated to my PR. Any idea how to fix them?

**Comment by rmg** [comment]
Interesting error:
```
Error: ENOTEMPTY, directory not empty '/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/test/sandbox/9000/.strong-pm/work'
      at Error (native)
      at Object.fs.rmdirSync (fs.js:711:18)
      at rmkidsSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:328:11)
      at rmdirSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:318:7)
      at rimrafSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:289:9)
      at /mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:326:5
      at Array.forEach (native)
      at rmkidsSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:325:26)
      at rmdirSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:318:7)
      at rimrafSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:289:9)
      at /mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:326:5
      at Array.forEach (native)
      at rmkidsSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:325:26)
      at rmdirSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:318:7)
      at rimrafSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:289:9)
      at /mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:326:5
      at Array.forEach (native)
      at rmkidsSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:325:26)
      at rmdirSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:318:7)
      at Function.rimrafSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/node_modules/rimraf/rimraf.js:289:9)
      at Object.removeSync (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/fs-extra/lib/remove.js:4:17)
      at removeSandBox (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/test/helpers.js:47:6)
      at Context.<anonymous> (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/test/manager-host.test.js:62:5)
      at Hook.Runnable.run (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/mocha/lib/runnable.js:233:15)
      at next (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/mocha/lib/runner.js:268:10)
      at Immediate._onImmediate (/mnt/jenkins/workspace/strong-mesh-client.pr/4a890f47/node_modules/mocha/lib/runner.js:289:5)
      at processImmediate [as _immediateCallback] (timers.js:358:17)
```

Perhaps something is still running and creating files in that directory when mocha moves on to the `after` function?

**Comment by ritch** [comment]
It may be the ordering which we are running the sandbox cleanup. I tried switching it but was getting some obscure PM errors related to removing the sandbox directory (eg. `process.cwd()` throwing).

**Pull request merged**